tez-ext-service-tests/src/test/java/org/apache/tez/tests/ExternalTezServiceTestHelper.java

```diff
@@ -78,10 +78,17 @@ public ExternalTezServiceTestHelper(String testRootDir) throws
     for (Map.Entry<String, String> entry : tezCluster.getConfig()) {
       clusterConf.set(entry.getKey(), entry.getValue());
     }
+
     long jvmMax = Runtime.getRuntime().maxMemory();
+    long availableMemory;
+    if ((jvmMax * 0.5d) >= Integer.MAX_VALUE) {
```

Comment:
could this be simplified as:
```
long availableMemory = (long) Math.min(jvmMax * 0.5d, Integer.MAX_VALUE);
```
also, what is this for? how is it related to JDK17?

Reply:
with JDK17 the available memory is coming big number from (Runtime.getRuntime().maxMemory();)
So in our code we are converting the long value to integer and we are facing numberFormat exceptions with jdk17. So i added check, >>long availableMemory = (long) Math.min(jvmMax * 0.5d, Integer.MAX_VALUE);
Anyway this value we are using in creating MiniTezTestServiceCluster.

Reply:
maxMemory() is behaving differently with jdk17 compare to old jdk8.
The number is: 8589934592 bigger than Integer.MAX_VALUE(2147483647) which is causing Number format exceptions while converting long to int. 
I will add comment to the code lines. Thank you !